x-pack/legacy/plugins/uptime/server/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -25,11 +20,22 @@ export class UMKibanaBackendFrameworkAdapter implements UMBackendFrameworkAdapte
     this.plugins = plugins;
   }
 
-  public registerRoute<
-    RouteRequest extends UMFrameworkRequest,
-    RouteResponse extends UMFrameworkResponse
-  >(route: UMFrameworkRouteOptions<RouteRequest, RouteResponse>) {
-    this.server.route(route);
+  public registerRoute({ handler, method, options, path, validate }: UMRouteDefinition) {
+    const routeDefinition = {
+      path,
+      validate,
+      options,
+    };
+    switch (method) {
+      case 'GET':
+        this.server.route.get(routeDefinition, handler);
+        break;
+      case 'POST':
+        this.server.route.post(routeDefinition, handler);
+        break;
```

Comment:
IRouter currently only provides per http method APIs to register routes. 

Maybe we should something like `<T>route(method: <T extends RouteMethod>, registrar: RouteRegistrar<T>)` to address that kind of usages? It's pretty commun in router APIs. 

WDYT @restrry

Reply:
We were going to switch to this API in https://github.com/elastic/kibana/issues/44620#issue-488206019
Although we downgraded this issue since then. We can discuss if we want to implement this API earlier.